src/mongo/db/curop.h

```diff
@@ -305,6 +306,10 @@ class CurOp {
         if (_dbprofile <= 0)
             return false;
 
+        if (serverGlobalParams.rateLimit > 1 && _dbprofile >= 2 && ms < serverGlobalParams.slowMS) {
+            return _prng.nextCanonicalDouble() * serverGlobalParams.rateLimit < 1.0;
```

Comment:
Processing doubles is much slower than ints. Why not to operate on ints instead?
```
const RateLimitMax = 1000;
_prng.nextInt64(RateLimitMax) + 1 <= serverGlobalParams.rateLimit
```